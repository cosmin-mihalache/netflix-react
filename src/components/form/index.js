import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
