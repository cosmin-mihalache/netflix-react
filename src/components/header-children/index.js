import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
