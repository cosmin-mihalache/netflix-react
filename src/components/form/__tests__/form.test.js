import React from 'react';
import { render } from '@testing-library/react';
import { Form } from '../../../components';

jest.mock('react-router-dom');

describe('<Form />', () => {
  it('renders the <Form /> with populated data', () => {
    const { container, getByText, getByPlaceholderText } = render(
      <Form>
        <Form.Title>Sign In Now</Form.Title>
        <Form.Base>
          <Form.Input placeholder="Email Address" onChange={() => {}} />
          <Form.Input
            placeholder="Password"
            type="password"
            type="password"
            onChange={() => {}}
          />
          <Form.Submit type="submit" disabled>
            Sign In
          </Form.Submit>
        </Form.Base>
        <Form.Text>
          New to Netflix.<Form.Link to="/signup"> Sign up now.</Form.Link>
        </Form.Text>
        <Form.SmallText>
          This page is protected by Google reCAPTCHA.
        </Form.SmallText>
      </Form>
    );
    expect(getByText('Sign In Now')).toBeTruthy();
    expect(getByText('Sign In')).toBeTruthy();
    expect(getByText('Sign In').disabled).toBeTruthy();
    expect(
      getByText('This page is protected by Google reCAPTCHA.')
    ).toBeTruthy();
    expect(getByPlaceholderText('Email Address')).toBeTruthy();
    expect(getByPlaceholderText('Password')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders a <Form /> with an error', () => {
    const { container, getByText, queryByText } = render(
      <Form>
        <Form.Error>Your email address is already being used.</Form.Error>
        <Form.Submit type="submit">Sign In</Form.Submit>
      </Form>
    );

    expect(getByText('Your email address is already being used.')).toBeTruthy();
    expect(queryByText('Sign In').disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
