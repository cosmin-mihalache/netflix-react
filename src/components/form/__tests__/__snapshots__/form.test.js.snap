// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> renders a <Form /> with an error 1`] = `
<section
  class="form__Container-sc-1qmgofh-0 bPznDn"
>
  <div
    class="form__Error-sc-1qmgofh-2 fzCGYt"
  >
    Your email address is already being used.
  </div>
  <button
    class="form__Submit-sc-1qmgofh-7 jhzVJg"
    type="submit"
  >
    Sign In
  </button>
</section>
`;

exports[`<Form /> renders the <Form /> with populated data 1`] = `
<section
  class="form__Container-sc-1qmgofh-0 bPznDn"
>
  <h1
    class="form__Title-sc-1qmgofh-3 kojpRx"
  >
    Sign In Now
  </h1>
  <form
    class="form__Base-sc-1qmgofh-1 gHxwcz"
  >
    <input
      class="form__Input-sc-1qmgofh-6 iQeJOG"
      placeholder="Email Address"
    />
    <input
      class="form__Input-sc-1qmgofh-6 iQeJOG"
      placeholder="Password"
      type="password"
    />
    <button
      class="form__Submit-sc-1qmgofh-7 jhzVJg"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <h1
    class="form__Text-sc-1qmgofh-4 dcrbkK"
  >
    New to Netflix.
  </h1>
  <h2
    class="form__SmallText-sc-1qmgofh-5 hFwlqp"
  >
    This page is protected by Google reCAPTCHA.
  </h2>
</section>
`;
